### ISSUE TITLE
Switch off part count in BOM

### ISSUE THREAD

**Issue by technicbasics** [trevorsandy/lpub3d#766]
Would it be possible, without great effort, to switch off the counting function in the BOM so that each part could be displayed individually?

**Comment by trevorsandy** [comment]
> Would it be possible, without great effort, to switch off the counting function in the BOM so that each part could be displayed individually?

Off the cuff, I would say I don't think there would be a 'great' effort involved. In fact, the subModel module - that displays a preview of the current subModel - is derived from the PLI module. Basically, it is displaying one 'part', the subModel. To get, started, I don't think it would be too much effort to feed an instance of this module the raw list of parts.

Cheers,

**Comment by trevorsandy** [comment]
Actually, I did a quick, successful POC with 7 lines of code :smile:. So, indeed, not too much effort...

![NoBOMCount_1](https://github.com/user-attachments/assets/9bb31415-f0a5-4316-ac06-c1815fb1e1ef)

Cheers,

**Comment by technicbasics** [comment]
I (and some others in the LDraw community) would be delighted if this option were included in a future version of LPub3D.

**Comment by trevorsandy** [comment]
Looks like X-Mas has come early ! - this feature has been implemented.

BOM Global Settings - also available in PLI Global Settings
![NoBomCount_2](https://github.com/user-attachments/assets/d566e908-1159-4e6e-adcf-a9f07b78aaa9)

Part number instance count will not display when show individual part instance is enabled.
![NoBomCount_3](https://github.com/user-attachments/assets/f6449148-378d-436b-8db4-257c2bfb3566)

Cheers,

**Comment by jacovandermolen** [comment]
Nice. Now we can create knolling pictures of a model ;-)

**Comment by jacovandermolen** [comment]
Follow up: how to configure the BOM to display parts from the top view?

**Comment by trevorsandy** [comment]
> Follow up: how to configure the BOM to display parts from the top view?

Is this a question for me ? If yes, I don't understand. 'Top view' of what exactly ?

Cheers,

**Comment by jacovandermolen** [comment]
Now, of course, ideally the wheel covers, tires, wheels, and some other parts would need a different angle to mimic how you would knoll this in real life with physical bricks. However, that should be achievable using the PLI Orientation/Size Control File ;-)

**Comment by trevorsandy** [comment]
Excellent! I like your use case of overhead photography for this feature. If you care to share your knoll orientation for the various parts where a different angle from 90 is preferred, I will take a look at automating the knoll view as an option when generating a BOM.

Cheers,

**Comment by trevorsandy** [comment]
FYI - did you know you can use the blender export to create exploded views ?

![CAR_Exploded_533x358](https://github.com/user-attachments/assets/f77ef2e5-a6af-4f13-a2d7-6366f6ecb12c)

See _car-exploded.ldr_ example
<details>

```forth
0 Car Exploded
0 Name: car-exploded.ldr
0 Author: 
0 !LDRAW_ORG Unknown

0 BFC CERTIFY CCW

1 7 0 21.23 -164.42 1 0 0 0 1 0 0 0 1 4600.dat
1 7 0 21.23 171.42 1 0 0 0 1 0 0 0 1 4600.dat
1 7 0 -68.33 -248.38 1 0 0 0 1 0 0 0 1 3938.dat
1 4 0 -1.16 143.44 0 0 1 0 1 0 -1 0 0 3021.dat
1 0 0 21.23 3.5 1 0 0 0 1 0 0 0 1 3031.dat
1 4 0 -1.16 -164.42 0 0 1 0 1 0 -1 0 0 3020.dat
1 4 83.96 -68.33 87.46 1 0 0 0 1 0 0 0 1 3005.dat
1 4 -83.96 -68.33 87.46 1 0 0 0 1 0 0 0 1 3005.dat
1 7 0 -68.33 -248.38 1 0 0 0 1 0 0 0 1 3937.dat
1 4 0 -45.94 -164.42 1 0 0 0 1 0 0 0 1 3788.dat
1 4 0 -45.94 171.42 1 0 0 0 1 0 0 0 1 3788.dat
1 4 0 -1.16 255.38 1 0 0 0 1 0 0 0 1 3710.dat
1 39 0 -225.06 -80.46 1 0 0 0 1 0 0 0 1 3823.dat
1 39 0 -225.06 199.41 -1 0 0 0 1 0 0 0 -1 3823.dat
1 4 83.96 -1.16 87.46 1 0 0 0 1 0 0 0 1 3024.dat
1 4 -83.96 -1.16 87.46 1 0 0 0 1 0 0 0 1 3024.dat
1 0 0 -225.06 87.46 1 0 0 0 1 0 0 0 1 4214.dat
1 0 0 21.23 -248.38 1 0 0 0 1 0 0 0 1 4315.dat
1 0 0 21.23 255.38 -1 0 0 0 1 0 0 0 -1 4315.dat
1 0 83.96 -90.72 115.45 0 0 1 0 1 0 -1 0 0 3023.dat
1 0 -83.96 -90.72 115.45 0 0 1 0 1 0 -1 0 0 3023.dat
1 1 0 -1.16 3.5 1 0 0 0 1 0 0 0 1 4079.dat
1 4 83.96 -68.33 -80.46 1 0 0 0 1 0 0 0 1 3822.dat
1 4 0 -90.72 -220.4 1 0 0 0 1 0 0 0 1 3068b.dat
1 0 83.96 -90.72 199.41 1 0 0 0 1 0 0 0 1 3024.dat
1 0 -83.96 -90.72 199.41 1 0 0 0 1 0 0 0 1 3024.dat
1 0 -83.96 38.02 171.42 0 0 1 0 1 0 -1 0 0 3641.dat
1 0 83.96 38.02 171.42 0 0 -1 0 1 0 1 0 0 3641.dat
1 0 -83.96 38.02 -164.42 0 0 1 0 1 0 -1 0 0 3641.dat
1 0 83.96 38.02 -164.42 0 0 -1 0 1 0 1 0 0 3641.dat
1 4 -83.96 -68.33 -80.46 1 0 0 0 1 0 0 0 1 3821.dat
1 4 83.96 -68.33 -164.42 0 0 1 0 1 0 -1 0 0 3023.dat
1 4 -83.96 -68.33 -164.42 0 0 1 0 1 0 -1 0 0 3023.dat
1 4 83.96 -90.72 -220.4 0 0 1 0 1 0 -1 0 0 3023.dat
1 4 -83.96 -90.72 -220.4 0 0 1 0 1 0 -1 0 0 3023.dat
1 4 83.96 -90.72 -108.45 0 0 1 0 1 0 -1 0 0 3023.dat
1 4 -83.96 -90.72 -108.45 0 0 1 0 1 0 -1 0 0 3023.dat
1 0 0 -247.45 3.5 1 0 0 0 1 0 0 0 1 4213.dat
1 46 83.96 -1.16 -248.38 1 0 0 0 1 0 0 0 1 3024.dat
1 46 -83.96 -1.16 -248.38 1 0 0 0 1 0 0 0 1 3024.dat
1 46 83.96 -23.55 255.38 1 0 0 0 1 0 0 0 1 3024.dat
1 46 -83.96 -23.55 255.38 1 0 0 0 1 0 0 0 1 3024.dat
1 7 -83.96 38.02 171.42 0 0 1 0 1 0 -1 0 0 4624.dat
1 7 83.96 38.02 171.42 0 0 -1 0 1 0 1 0 0 4624.dat
1 7 -83.96 38.02 -164.42 0 0 1 0 1 0 -1 0 0 4624.dat
1 7 83.96 38.02 -164.42 0 0 -1 0 1 0 1 0 0 4624.dat
1 0 0 -247.45 171.42 1 0 0 0 1 0 0 0 1 3020.dat
1 4 0 -68.33 255.38 1 0 0 0 1 0 0 0 1 3004.dat
1 4 0 -90.72 -136.43 1 0 0 0 1 0 0 0 1 3004.dat
1 0 0 -90.72 255.38 1 0 0 0 1 0 0 0 1 3710.dat
1 46 83.96 -40.34 -287.56 1 0 0 0 0 -1 0 1 0 6141.dat
1 46 -83.96 -40.34 -287.56 1 0 0 0 0 -1 0 1 0 6141.dat
1 4 0 -23.55 -80.46 1 0 0 0 1 0 0 0 1 3829c01.dat
1 36 83.96 -45.94 255.38 1 0 0 0 1 0 0 0 1 3024.dat
1 36 -83.96 -45.94 255.38 1 0 0 0 1 0 0 0 1 3024.dat
1 4 83.96 -1.16 -24.49 0 0 1 0 1 0 -1 0 0 3623.dat
1 4 -83.96 -1.16 -24.49 0 0 1 0 1 0 -1 0 0 3623.dat
1 4 83.96 -68.33 199.41 0 0 1 0 1 0 -1 0 0 3623.dat
1 4 -83.96 -68.33 199.41 0 0 1 0 1 0 -1 0 0 3623.dat
1 4 83.96 -68.33 -248.38 1 0 0 0 1 0 0 0 1 4070.dat
1 4 -83.96 -68.33 -248.38 1 0 0 0 1 0 0 0 1 4070.dat
```

</details>

Cheers,

**Comment by jacovandermolen** [comment]
> Excellent! I like your use case of overhead photography for this feature. If you care to share your knoll orientation for the various parts where a different angle from 90 is preferred, I will take a look at automating the knoll view as an option when generating a BOM.

Will do! I will discuss with Johann too.

**Comment by technicbasics** [comment]
@jacovandermolen 
You also couls set the camera angels in the BOM setup menu

![grafik](https://github.com/user-attachments/assets/7c0eb453-1a13-4bb6-a6af-b1926d6c35d6)

Regards Johann

**Comment by technicbasics** [comment]
> FYI - did you know you can use the blender export to create exploded views ?
> 
> ![CAR_Exploded_533x358](https://private-user-images.githubusercontent.com/13388970/375361309-f77ef2e5-a6af-4f13-a2d7-6366f6ecb12c.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mjg1Nzc5MTYsIm5iZiI6MTcyODU3NzYxNiwicGF0aCI6Ii8xMzM4ODk3MC8zNzUzNjEzMDktZjc3ZWYyZTUtYTZhZi00ZjEzLWEyZDctNjM2NmY2ZWNiMTJjLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEwMTAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMDEwVDE2MjY1NlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTQyYmQ3ZjhlNmI5YzU1NjQ4MTBmNTM0ZTk0Yjc2NGNlN2M3ZTk4YWJlMmJkYmQxYTg3MDg0NmQ3YTIyNWZkNmEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.S90KWaX1fP4sH-kBGp4l0fPnb0DL47SrI35dLwNUfYY)
> 
> See _car-exploded.ldr_ example
> 

> 
> Cheers,

Wich blender version is needed or supported to do this?

**Comment by trevorsandy** [comment]
> Which blender version is needed or supported to do this ?

Actually, I can't remember which version introduced the ability to [explode an assembly](https://www.youtube.com/watch?v=_9GYrjhjtdo&ab_channel=TheCGEssentials) but it's a pretty old one. However, more importantly, is the Blender version supported by the LPub3D (and LeoCAD) [BlenderLDrawRender](https://github.com/trevorsandy/blenderldrawrender) python LDraw Import/Export add-on I created which is v2.8.2 to at least v4.2.1 - the version I used for my last update.

The steps to reproduce are:
- Load your model into LPub3D
- Select the _Visual Editor_=>_Open In Blender_ menu item...
- Select Settings and follow the prompts to download and setup the LDraw add-on if not already installed - be sure to select MM as your LDraw importer as this submodule has the capability to export LDraw files.
- After successful addon-setup, click the _Open In Blender_ button
- If all is properly configured, Blender will launch with your model loaded
- Follow the instructions in the video [explode an assembly](https://www.youtube.com/watch?v=_9GYrjhjtdo&ab_channel=TheCGEssentials)
- Export your model
- Import your exported model file and voila!

Cheers,
 
Cheers,
